### PR TITLE
[MM-45423]: re-added save option to post-dotmenu

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
re-added the save/unsave option to the post dotmenu

#### Ticket Link
[MM-45423](https://mattermost.atlassian.net/browse/MM-45423)

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by matthewbirtch — APPROVED** [review]
LGTM

**Comment by jgilliam17** [comment]
Apologies, requesting changes after posting approval 🤦🏽‍♀️ - We have a test for Copy menu option that you can use as a guideline `e2e/cypress/tests/integration/messaging/copy_post_text_spec.js`, can you please add something similar for Save? Thanks 🙂

**Review by hmhealey — APPROVED** [review]

**Review by jfrerich — APPROVED** [review]
LGTM!

**Comment by michelengelen** [comment]
@jgilliam17 I have added tests for saving and unsaving a post.

I also did change add a new verification function, moved it to the post.js file and used it in saving spec and the post-pre-header spec (where i got most of the code from.

cc @saturninoabril

**Comment by jgilliam17** [comment]
Thanks @michelengelen for the test. 
Can you please fix failed checks, so we can trigger e2es

File: e2e/cypress/tests/integration/messaging/save_post_spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+describe('Save Post', () => {
+    before(() => {
+        cy.apiInitSetup().then(({channelUrl}) => {
+            // # Go to test channel
+            cy.visit(channelUrl);
+        });
+    });
+
+    it('MM-Txxxx Save menu item should save post', () => {
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/messaging/save_post_spec.js:22]:
```suggestion
    it('MM-T4948_1 Save menu item should save post', () => {
```

File: e2e/cypress/tests/integration/messaging/save_post_spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+            // * Assert the post pre-header is displayed and works as expected
+            verifySavedPost(postId, message);
+
+            // # Remove post from saved by clicking the menu item in the dotmenu
+            cy.uiClickPostDropdownMenu(postId, 'Remove from Saved');
+        });
+    });
+
+    it('MM-Txxxx Save hotkey should save post', () => {
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/messaging/save_post_spec.js:45]:
```suggestion
    it('MM-T4948_2 Save hotkey should save post', () => {
```

File: e2e/cypress/tests/integration/messaging/save_post_spec.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        cy.get('@postId1').then((postId) => {
+            // # Save message by clicking the menu item in the dotmenu
+            cy.uiClickPostDropdownMenu(postId, 'Save');
+
+            // * Assert the post pre-header is displayed and works as expected
+            verifySavedPost(postId, message);
+
+            // # Remove post from saved by clicking the menu item in the dotmenu
+            cy.uiClickPostDropdownMenu(postId, 'Remove from Saved');
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/messaging/save_post_spec.js:41]: Can we add a line here to verify post was unsaved?

File: e2e/cypress/tests/integration/messaging/save_post_spec.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            cy.clickPostDotMenu(postId, 'CENTER');
+
+            // # Press s to save
+            cy.get('body').type('s');
+
+            // * Assert the post pre-header is displayed and works as expected
+            verifySavedPost(postId, message);
+
+            // # Press s again to unsave
+            cy.get('body').type('s');
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/messaging/save_post_spec.js:67]: same here, can we check if it was unsaved successfully

**Review by jgilliam17 — CHANGES_REQUESTED** [review]
Adding few test id suggestions

File: e2e/cypress/tests/integration/messaging/save_post_spec.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [#] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/messaging/save_post_spec.js:9]:
```suggestion

// Stage: @prod
```

**Review by jgilliam17 — COMMENTED** [review]
Can we please add this tag too? 
Sorry I missed this, I just realized while looking at the report and could't find the new tests there.

**Comment by jgilliam17** [comment]
@michelengelen apologies for delay, I'm catching up after time off 
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/223871-3147883-onprem-ent-webapp-pr-10701-mattermost/mm-ee-test3147883/mochawesome.html)shows new tests are failing (MM-T4948)

**Comment by jgilliam17** [comment]
Checking why cypress run [failed](https://automation-dashboard.vercel.app/cycles/3c1242be-ca8a-4ae3-9029-ca3bdcf58ebf) to run

**Comment by jgilliam17** [comment]
Thanks @michelengelen - [E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/229714-84be26d-onprem-ent-webapp-pr-10701-mm-ee-test:84be26d/mochawesome.html) looks good.
Screen Shot 2022-08-15 at 3 45 09 PM

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**